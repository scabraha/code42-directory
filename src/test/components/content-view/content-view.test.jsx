import React from 'react';
import renderer from 'react-test-renderer';
import ContentView from '../../../components/content-view/content-view';

jest.mock('../../../components/member-details/member-details', () => 'MemberDetails');

describe('Content View', () => {
  it('Renders the view without member details and ensures logo and welcome message show', () => {
    const tree = renderer.create(
      <ContentView memberDetailsRequestStatus="NOT_STARTED" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view while loading member details and ensures logo and loading message show', () => {
    const tree = renderer.create(
      <ContentView memberDetailsRequestStatus="LOADING" selectedMemberId="testId" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with member details and ensures details show', () => {
    const details = {
      avatar_url: 'testURL',
      name: 'testName',
      created_at: '6/05/2001',
      location: 'USA',
      email: 'test@test.com'
    }
    const tree = renderer.create(
      <ContentView 
        memberDetailsRequestStatus="SUCCESS" 
        selectedMemberId="testId" 
        memberDetails={details}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

