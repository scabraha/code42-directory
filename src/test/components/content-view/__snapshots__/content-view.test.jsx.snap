// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content View Renders the view while loading member details and ensures logo and loading message show 1`] = `
<div
  className="content-view"
>
  <div
    className="content-view-header"
  >
    <img
      alt="avatar"
      className="content-view-image"
      src="logo.svg"
    />
    <h2
      className="md-headline"
    >
      Loading testId's details
    </h2>
  </div>
</div>
`;

exports[`Content View Renders the view with member details and ensures details show 1`] = `
<div
  className="content-view"
>
  <div
    className="content-view-header"
  >
    <img
      alt="avatar"
      className="content-view-image"
      src="testURL"
    />
    <h2
      className="md-headline"
    >
      testName
    </h2>
  </div>
  <MemberDetails
    email="test@test.com"
    joinDate="6/05/2001"
    location="USA"
  />
</div>
`;

exports[`Content View Renders the view without member details and ensures logo and welcome message show 1`] = `
<div
  className="content-view"
>
  <div
    className="content-view-header"
  >
    <img
      alt="avatar"
      className="content-view-image"
      src="logo.svg"
    />
    <h2
      className="md-headline"
    >
      Welcome to the Code42 GitHub Directory
    </h2>
  </div>
  <p
    className="content-view-body"
  >
    Select a member to learn more about them
  </p>
</div>
`;
